packages/eslint-plugin-internal/src/rules/all-bases.js

```diff
@@ -23,8 +23,7 @@ module.exports = {
   'import/no-extraneous-dependencies': [
     'error',
     {
-      devDependencies: ['**/packages/common/**', '**/packages/eslint-plugin-internal/**'],
-      optionalDependencies: false,
+      devDependencies: true,
```

Comment:
I've discussed this with @andrewballantyne - this setting allows using dev dependencies in all monorepo packages.